deployment/hasura/migrations/AerieMerlin/23_activity_directive_changelog/down.sql

```diff
@@ -0,0 +1,1006 @@
+    if not exists(select *
+                  from public.plan p
+                  where p.id = _plan_id_supplying and p.owner = _user) then
+      if not exists(select *
+                    from public.plan_collaborators pc
+                    where pc.plan_id = _plan_id_supplying and pc.collaborator = _user) then
+        raise insufficient_privilege
+          using message = 'Cannot run '''|| _function ||''': '''|| _user ||''' is not PLAN_OWNER_COLLABORATOR on Source Plan '
+                          || _plan_id_supplying ||' ('|| _supplying_plan_name ||').';
+      end if;
+    end if;
+
+  -- PLAN_OWNER_TARGET: The user must be the Owner of the Receiving Plan.
+  elseif _permission = 'PLAN_OWNER_TARGET' then
+    if not exists(select *
+                  from public.plan p
+                  where p.id = _plan_id_receiving and p.owner = _user) then
+      raise insufficient_privilege
+        using message = 'Cannot run '''|| _function ||''': '''|| _user ||''' is not PLAN_OWNER on Target Plan '
+                          || _plan_id_receiving ||' ('|| _receiving_plan_name ||').';
+    end if;
+
+  -- PLAN_COLLABORATOR_TARGET: The user must be a Collaborator of the Receiving Plan.
+  elseif _permission = 'PLAN_COLLABORATOR_TARGET' then
+    if not exists(select *
+                  from public.plan_collaborators pc
+                  where pc.plan_id = _plan_id_receiving and pc.collaborator = _user) then
+      raise insufficient_privilege
+        using message = 'Cannot run '''|| _function ||''': '''|| _user ||''' is not PLAN_COLLABORATOR on Target Plan '
+                          || _plan_id_receiving ||' ('|| _receiving_plan_name ||').';
+    end if;
+
+  -- PLAN_OWNER_COLLABORATOR_TARGET: The user must be either the Owner or a Collaborator of the Receiving Plan.
+  elseif _permission = 'PLAN_OWNER_COLLABORATOR_TARGET' then
+    if not exists(select *
+                  from public.plan p
+                  where p.id = _plan_id_receiving and p.owner = _user) then
+      if not exists(select *
+                    from public.plan_collaborators pc
+                    where pc.plan_id = _plan_id_receiving and pc.collaborator = _user) then
+        raise insufficient_privilege
+          using message = 'Cannot run '''|| _function ||''': '''|| _user ||''' is not PLAN_OWNER_COLLABORATOR on Target Plan '
+                          || _plan_id_receiving ||' ('|| _receiving_plan_name ||').';
+      end if;
+    end if;
+  end if;
+end
+$$;
+
+-- drop activity_directive_changelog table and functions
+drop table activity_directive_changelog;
+
+drop trigger store_activity_directive_change_trigger on activity_directive;
+drop function store_activity_directive_change;
+drop function delete_min_activity_directive_revision;
+drop function hasura_functions.restore_activity_changelog(
+  _plan_id integer,
+  _activity_directive_id integer,
+  _revision integer,
+  hasura_session json);
```

Comment:
The correct order for this section is:
```suggestion
drop function hasura_functions.restore_activity_changelog(
  _plan_id integer,
  _activity_directive_id integer,
  _revision integer,
  hasura_session json);
drop trigger delete_min_activity_directive_revision_trigger on activity_directive_changelog;
drop function delete_min_activity_directive_revision;
drop trigger store_activity_directive_change_trigger on activity_directive;
drop function store_activity_directive_change;

comment on table activity_directive_changelog is null;
drop table activity_directive_changelog;
```

Additionally, the trigger `delete_min_activity_directive_revision_trigger` isn't being dropped here (the suggestion includes it).

Reply:
Dropping the table automatically drops all associated triggers as well as comments.